pkg/store/bucket.go

```diff
@@ -2397,16 +2454,19 @@ func ExpandPostingsWithContext(ctx context.Context, p index.Postings) (res []sto
 // If addAll is not set: Merge of postings for "addKeys" labels minus postings for removeKeys labels
 // This computation happens in ExpandedPostings.
 type postingGroup struct {
-	addAll     bool
-	name       string
-	addKeys    []string
-	removeKeys []string
+	addAll      bool
+	name        string
+	matchers    []*labels.Matcher
```

Comment:
Do we need to worry about the matchers now on the `merge` function?
Ex: `cpu_utilization{pod="a.+", pod!="aaabb"}` will be converted in one `postingGroup` but i dont see we are merging those matchers?


https://github.com/thanos-io/thanos/blob/266a76018aa39d104833095e00c1954a24ec3873/pkg/store/bucket.go#L2415

Reply:
I merged the matchers in https://github.com/thanos-io/thanos/pull/6465/files#diff-3e2896fafa6ff73509c77df2c4389b68828e02575bb4fb78b6c34bcfb922a7ceR2625

Reply:
oh ok..

Would it make more sense to add those matchers on the merge function? cause now you have a resulting merged posting group that is incomplete?

Reply:
Good catch. I don't know if there is a good way to merge matchers as it requires additional sorting and probably a map to dedup.

https://github.com/thanos-io/thanos/pull/6465/commits/27d04c85e7a7d0effb02febf684c10c3e479ed6c I have changed the method to only merge keys